src/__experimental__/components/decimal/decimal.d.ts

```diff
@@ -0,0 +1,34 @@
+import * as React from "react";
+
+export interface DecimalProps {
+  /** The default value alignment on the input */
+  align?: string;
+  /** The decimal precision of the value in the input */
+  precision?: number;
+  /** The width of the input as a percentage */
+  inputWidth?: number;
+  /** If true, the component will be read-only */
+  readOnly?: boolean;
+  /** The default value of the input if it's meant to be used as an uncontrolled component */
+  defaultValue?: string;
+  /** The value of the input if it's used as a controlled component */
+  value?: string;
+  /** Handler for change event if input is meant to be used as a controlled component */
+  onChange?: (ev: React.ChangeEvent<HTMLElement>) => void;
```

Comment:
I think this and `onBlur` need to be `<HTMLInputElement>` so that the consumers can access target.value.